src/pocketmine/entity/Minecart.php

```diff
@@ -37,19 +38,36 @@ class Minecart extends Vehicle{
 	const TYPE_CHEST = 2;
 	const TYPE_HOPPER = 3;
 	const TYPE_TNT = 4;
+	
+	const STATE_INITIAL = 0;
+	const STATE_ON_RAIL = 1;
+	const STATE_OFF_RAIL = 2;
 
-	public $height = 0.9;
-	public $width = 1.1;
+	public $height = 0.7;
+	public $width = 0.98;
 
 	public $drag = 0.1;
 	public $gravity = 0.5;
 
 	public $isMoving = false;
 	public $moveSpeed = 0.4;
-
+	
+	private $onRail = Minecart::STATE_INITIAL;
+	private $direction = -1;
+	private $moveVector = [];
+	private $requestedPosition = null;
+	
 	public function initEntity(){
 		$this->setMaxHealth(1);
 		$this->setHealth($this->getMaxHealth());
+		$this->moveVector[Entity::NORTH] = new Vector3(-1, 0, 0);
+		$this->moveVector[Entity::SOUTH] = new Vector3(1, 0, 0);
+		$this->moveVector[Entity::EAST] = new Vector3(0, 0, -1);
+		$this->moveVector[Entity::WEST] = new Vector3(0, 0, 1);
+		$this->moveVector[Entity::NORTH] = new Vector3(-1, 0, 0);
+		$this->moveVector[Entity::SOUTH] = new Vector3(1, 0, 0);
+		$this->moveVector[Entity::EAST] = new Vector3(0, 0, -1);
+		$this->moveVector[Entity::WEST] = new Vector3(0, 0, 1);
```

Comment:
Sorry, thought I had cleaned things up, but missed this.  I will fix.   At one point I was going to have UpNorth, ..., DownNorth, ... but then didn't and forgot to remove the copied lines.